index.bs

```diff
@@ -8953,35 +8953,29 @@ Constructors</h4>
 	: <dfn>PeriodicWave(context, options)</dfn>
 	::
 		<div algorithm="construct periodic wave">
-			1. Let <var>p</var> be a new {{PeriodicWave}} object. Let
-				<dfn attribute for="PeriodicWave">[[associated context]]</dfn> be a reference to the
-				{{BaseAudioContext}} passed as first argument of this
-				constructor.
-
-			2. <span class="synchronous">If the {{PeriodicWaveOptions/real}} and {{PeriodicWaveOptions/imag}} parameters of
-				the {{PeriodicWaveOptions}} are not of the same length, an
-				{{IndexSizeError}} exception MUST be thrown.</span>
-
-			3. If {{PeriodicWave/PeriodicWave(context, options)/options!!argument}} has not been passed in, let
-				<dfn attribute for="PeriodicWave">\[[real]]</dfn> and <dfn attribute for="PeriodicWave">\[[imag]]</dfn> be two internal
-				slots of type {{Float32Array}} and length 2. Set the
-				second element of the {{[[imag]]}} array to 1.
-
-				Note: When setting this {{PeriodicWave}} on an
-				{{OscillatorNode}}, this is equivalent to using the
-				built-in type <code>"sine"</code>.
-
-			4. Otherwise, let {{[[real]]}} and
-				{{[[imag]]}} be two internal slots of type
-				{{Float32Array}}, both of length equal to the maximum of the
-				lengths of the {{PeriodicWaveOptions/real}} and {{PeriodicWaveOptions/imag}} attributes of
-				the {{PeriodicWaveOptions}} passed in.
-				<a href="https://heycam.github.io/webidl/#dfn-get-buffer-source-copy">
-				Make a copy</a> of those arrays into their respective
-				internal slots.
-
-			5. Let <dfn attribute for="PeriodicWave">\[[normalize]]</dfn> be an internal slot of the
-				{{PeriodicWave}} that is initialized to the inverse of the
+			1. Let <var>p</var> be a new {{PeriodicWave}} object. Let <dfn attribute for="PeriodicWave">\[[real]]</dfn> and <dfn attribute for="PeriodicWave">\[[imag]]</dfn> be two internal slots of type {{Float32Array}}, and let <dfn attribute for="PeriodicWave">\[[normalize]]</dfn> be an internal slot.
+
+			1. Process {{PeriodicWave/PeriodicWave(context, options)/options!!argument}} according to one of the following cases:
+				1. If both {{PeriodicWaveOptions/real|options.real}} and {{PeriodicWaveOptions/imag|options.imag}} are given
+					1. <span class="synchronous">If the lengths are different or if either length is less than 2, throw an {{IndexSizeError}} and abort this algorithm.</span>
+					1. Set {{[[real]]}} and {{[[imag]]}} to new arrays with a length of {{PeriodicWaveOptions/real|options.real}}.
```

Comment:
Are these objects observable later as actual objects in JS?  That is, does is matter what Realm is used for these allocations?

Also "with a length of" might be better as "with the same length as", like the other cases.

Reply:
The internal slots are internal and aren't ever exposed to JS in any way.